### PR TITLE
Dhp 94 authenticate with VA

### PR DESCRIPTION
**Pull request by TravisGM**
Description
In order for a Veteran to connect their wearable devices, so that we may display their patient generated data (PGD) for clinicians, Veterans need to be able to authenticate their credentials. After authenticating their credentials, they need to be redirected to the DHP landing page, which will eventually show the devices they've connected.

Original issue(s)
department-of-veterans-affairs/va.gov-team#0000

Testing done
Unit testing has been done on code that has been added.

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/dhp-connected-devices/tests/dhp-consent-flow.cypress.spec.js
Comment (yonidoronpeters) [inline src/applications/dhp-connected-devices/tests/dhp-consent-flow.cypress.spec.js:4]: See the warnings on this file: "Cypress E2E tests must include at least one axeCheck call". Not sure what this means, but seems to be related to linting rules

File: src/applications/dhp-connected-devices/containers/App.jsx
Comment (yonidoronpeters) [inline src/applications/dhp-connected-devices/containers/App.jsx:49]: Does it make sense to extract this part into a component?

File: src/applications/dhp-connected-devices/containers/App.jsx
Comment (yonidoronpeters) [inline src/applications/dhp-connected-devices/containers/App.jsx:77]: Same question as above - does it make sense to extract this to a component?

File: src/platform/site-wide/user-nav/components/SearchHelpSignIn.jsx
Comment (yonidoronpeters) [inline src/platform/site-wide/user-nav/components/SearchHelpSignIn.jsx:68]: Do we need to get an approval from the VSP team for changes outside the dhp-connected-devices app?

**Review by yonidoronpeters — COMMENTED** [review]
Overall looks great - added a few questions

File: src/applications/dhp-connected-devices/tests/App.spec.jsx
Comment (TravisGM) [reply src/applications/dhp-connected-devices/tests/App.spec.jsx:12]: Wow, good catch! Will fix that

File: src/applications/dhp-connected-devices/tests/dhp-consent-flow.cypress.spec.js
Comment (TravisGM) [reply src/applications/dhp-connected-devices/tests/dhp-consent-flow.cypress.spec.js:4]: I saw some of the other tests including `axeCheck`, we'll work on that. Thanks for pointing that out

File: src/applications/dhp-connected-devices/containers/App.jsx
Comment (TravisGM) [reply src/applications/dhp-connected-devices/containers/App.jsx:49]: @yonidoronpeters it might, but if you notice on line 43 we're using a component called `<UserNav />`. Are you suggesting we might be able to pull all of this code into a component, including the component on line 43?

File: src/platform/site-wide/user-nav/components/SearchHelpSignIn.jsx
Comment (TravisGM) [reply src/platform/site-wide/user-nav/components/SearchHelpSignIn.jsx:68]: @yonidoronpeters I don't believe so. But I'll double check, thanks for that

File: src/applications/dhp-connected-devices/containers/App.jsx
Comment (reblg) [reply src/applications/dhp-connected-devices/containers/App.jsx:49]: You can nest them I think! Good call Yoni

File: src/applications/dhp-connected-devices/tests/dhp-consent-flow.cypress.spec.js
Comment (reblg) [reply src/applications/dhp-connected-devices/tests/dhp-consent-flow.cypress.spec.js:4]: Where were you seeing the warnings @yonidoronpeters

Comment (TravisGM) [reply src/applications/dhp-connected-devices/tests/dhp-consent-flow.cypress.spec.js:4]: We actually attempted to run an `axeCheck` and the accessibility issues were related to modals & components that we didn't make. A lot of the issues were around using the same ID for various HTML elements.

**Comment by TravisGM** [comment]
@department-of-veterans-affairs/frontend-review-group would you mind taking a look at file `src/platform/site-wide/user-nav/components/SearchHelpSignIn.jsx`? We made a slight change to it, to allow a custom text to be displayed on the `Sign in` button. The custom text we'd like to use is `Sign in or create an account`

File: src/applications/dhp-connected-devices/tests/dhp-consent-flow.cypress.spec.js
Comment (yonidoronpeters) [reply src/applications/dhp-connected-devices/tests/dhp-consent-flow.cypress.spec.js:4]: @reblg seeing it in the PR under the Files changed tab
Screen Shot 2022-03-29 at 11 51 26 AM

File: src/platform/site-wide/user-nav/components/SearchHelpSignIn.jsx
Comment (reblg) [reply src/platform/site-wide/user-nav/components/SearchHelpSignIn.jsx:68]: @theodur This is what we are referring to in slack

Comment (theodur) [reply src/platform/site-wide/user-nav/components/SearchHelpSignIn.jsx:68]: Any changes to files outside your team's application will require approval from the [codeowners](https://github.com/department-of-veterans-affairs/vets-website/blob/master/.github/CODEOWNERS#L23). In this case that would be @department-of-veterans-affairs/vsp-identity

**Comment by theodur** [comment]
> @department-of-veterans-affairs/frontend-review-group would you mind taking a look at file `src/platform/site-wide/user-nav/components/SearchHelpSignIn.jsx` in our PR? We made a slight change to it, to allow a custom text to be displayed on the `Sign in` button. The custom text we'd like to use is `Sign in or create an account`

I added @department-of-veterans-affairs/vsp-identity as a reviewer, since they're the [codeowner](https://github.com/department-of-veterans-affairs/vets-website/blob/master/.github/CODEOWNERS#L23) of `src/platform/site-wide/user-nav/components/SearchHelpSignIn.jsx`. I'm not sure why they weren't automatically added as a reviewer

**Review by reblg — APPROVED** [review]

File: src/applications/dhp-connected-devices/containers/App.jsx
Comment (yonidoronpeters) [inline src/applications/dhp-connected-devices/containers/App.jsx:32]: Loving this refactor ❤️

Comment (yonidoronpeters) [reply src/applications/dhp-connected-devices/containers/App.jsx:32]: One minor nitpick - Code comments are probably not needed here and usually become outdated quickly

File: src/applications/dhp-connected-devices/containers/App.jsx
Comment (yonidoronpeters) [inline src/applications/dhp-connected-devices/containers/App.jsx:33]: ❤️

**Review by yonidoronpeters — APPROVED** [review]
Loving the refactor!

Based on @theodur we'll need an approval from the vsp-identity team

**Review by SNKyle — APPROVED** [review]

**Review by Kwoly — APPROVED** [review]
lgtm from an Identity team perspective

**Review by theodur — APPROVED** [review]

**Pull request merged**
